library/local_semodule.py

```diff
@@ -0,0 +1,367 @@
+
+    :type module: AnsibleModule
+    :param module: Ansible module
+
+    :type path: str
+    :param name: a local module file (either .cil or .pp) to be installed on a node
+
+    :type priority: int
+    :param priority: SELinux module priority, 0 - use SELinux module store default (400)
+    """
+
+    path_checksum = "sha256:" + module.sha256(path)
+
+    filename = os.path.split(path)[1]
+    (name, ext) = os.path.splitext(filename)
+
+    if ext == ".cil":
+        cil = 1
+    elif ext == ".pp":
+        cil = 0
+    else:
+        raise Exception("Only .cil and .pp files are supported")
+
+    r, modkey = semanage.semanage_module_key_create(sh)
+    if r != 0:
+        raise Exception(r)
+
+    r = semanage.semanage_module_key_set_name(sh, modkey, name)
+    if r != 0:
+        raise Exception(r)
+
+    if priority == 0:
+        r, modinfo, num_modules = semanage.semanage_module_list(sh)
+        for n in range(num_modules):
+            m = semanage.semanage_module_list_nth(modinfo, n)
+
+            r, m_name = semanage.semanage_module_info_get_name(sh, m)
+            if name == m_name:
+                r, m_prio = semanage.semanage_module_info_get_priority(sh, m)
+                priority = m_prio
+                break
+    if priority != 0:
+        r = semanage.semanage_module_key_set_priority(sh, modkey, priority)
+        if r != 0:
+            raise Exception(r)
+
+    try:
+        r, module_checksum, _len = semanage.semanage_module_compute_checksum(
+            sh, modkey, cil
+        )
+
+    except Exception:
+        r = 0
+        module_checksum = ""
+
+    if r != 0:
+        raise Exception(r)
+
+    result = {
+        "changed": False,
+        "name": name,
+        "priority": priority,
+        "state": "enabled",
+    }
+
+    if path_checksum == module_checksum:
+        return result
+
+    if priority != 0:
+        semanage.semanage_set_default_priority(sh, priority)
+
+    semanage.semanage_module_install_file(sh, path)
+    semanage.semanage_commit(sh)
+
+    result["changed"] = True
+
+    return result
+
+
+def semodule_enable(module, name, enabled, sh):
+    """Enable or install a local policy module
+
+    :type module: AnsibleModule
+    :param module: Ansible module
+
+    :type name: str
+    :param name: module name, used for enabling disabled modules, disabling enabled modules, removing modules
+
+    :type enabled: int
+    :param enabled: 0 - disable, 1 - enable
+    """
+    result = {
+        "changed": False,
+        "name": name,
+        "state": "enabled" if enabled else "disabled",
+    }
+
+    r, modkey = semanage.semanage_module_key_create(sh)
+    if r != 0:
+        raise Exception(r)
+
+    r = semanage.semanage_module_key_set_name(sh, modkey, name)
+    if r != 0:
+        raise Exception(r)
+
+    try:
+        r, m_enabled = semanage.semanage_module_get_enabled(sh, modkey)
+    except Exception:
+        raise ValueError("Could not find module {name}".format(name=name))
```

Comment:
Is there any information in the exception, like an error message, that might be helpful to display to the user, to try to fix the problem?